### ISSUE TITLE
Error using in V12

### ISSUE THREAD

**Issue by MichaelDBA** [credativ/pg_checksums#9]
I am able to execute pg_checksums for PG v11, but not for PG v12:

```
/usr/bin/pg_checksums -v
pg_checksums: error: pg_control CRC value is incorrect
```

**Comment by mbanck** [comment]
Are you using distribution package or building pg_checksums yourself? If the latter, did you rebuild it for v12?

Do you have `PGDATA` as environment variable? If not, I would rather expect `pg_checksums: error: no data directory specified` as error message

**Comment by MichaelDBA** [comment]
I built pg_checksums using the makefile that came with the downloaded git project.  I do have PGDATA set as an environment variable for both v11 and v12.  I set it before I run the pg_checksums command for each execution.

Do I need to set pg_config before doing the make?

**Comment by mbanck** [comment]
I would say it is safer to build a version of `pg_checksums` per major version, e.g. Debian keeps them in `/usr/lib/postgresql/<version>/bin/pg_checksums`. However, `pg_checksums` should error out more gracefully if that is not the case.

I would suggest you first try to recompile setting `pg_config` appropriateley, like `make PG_CONFIG=/path/to/your/v12/pg_config` and let me know if that solves the issue for V12.

**Comment by MichaelDBA** [comment]
I did a make for both v11 and v12 and saved their binary names as pg_checksums_v11 and pg_checksums_v12.  Everything works ok now as long as I use the separate binary for the target that matches the version number.  I cannot use the PG v11 pg_checksums binary with the PG v12 or I get the same error as before.  They have to be compiled using their respective pg_configs.

Is there a future fix so we don't have to keep doing that?

I'm using Ubuntu 16.04 LTS.

**Comment by mbanck** [comment]
I will work on a better check and error reporting, but I am quite wary of making `pg_checksums` more version independent. I discussed it with my colleagues a whlie back and we agreed it was not worth possible subtle version incompatibilities.

Note that apt.postgresql.org should ship `pg_checksums` packages for 16.04. I am not 100% sure on the v12 status but I changed the Debian/Ubuntu package such that the `postgresql-12-pg-checksums` package diverts away the upstream `pg_checksums` tool. There is currently (and probably for the future, but I can look into that) no `pg_wrapper` support, so you will have to run the appropriate binary in `/usr/lib/postgresql/<11/12>/bin/pg_checksums` manually based on major version.

**Comment by MichaelDBA** [comment]
Thanks for your response.   Yeah I can see your concern when it comes to checksums and PG major versions.  NO problemo. I will just keep separate binaries for separate major versions.  You might want to add this to your documentation since this is very important.

**Comment by mbanck** [comment]
This should be fixed by 5980214ae82b03ef32b90d7bde65abadf0a692f7

**Comment by mbanck** [comment]
This has been fixed in the 1.1 release
